### PR TITLE
FAI-13080 | FarosGraph query mode for Jira boards

### PR DESCRIPTION
**Pull request by dbruno21**
## Description

Added boolean parameter and logic to use Faros Graph for Jira boards selection (inclusions / exclusions)

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/jira-source/resources/spec.json [hunk 1]
```diff
@@ -253,6 +253,17 @@
         "type": "string",
         "title": "Organization ID",
         "description": "Atlassian organization ID for fetching teams. Required only when using Fetch Teams with Jira Cloud. See https://confluence.atlassian.com/jirakb/what-it-is-the-organization-id-and-where-to-find-it-1207189876.html for information on how to find it."
+      },
+      "query_mode": {
```
Comment (ypc-faros) [inline sources/jira-source/resources/spec.json:257]: I was never a fan of this enum :D 
Should we use a boolean toggle instead?

File: sources/jira-source/src/project-board-filter.ts [hunk 2]
```diff
@@ -100,9 +183,53 @@ export class ProjectBoardFilter {
     }
   }
 
+  private async loadBoards(): Promise<void> {
```
Comment (ypc-faros) [inline sources/jira-source/src/project-board-filter.ts:186]: We should move this to `faros-airbyte-common` (doesn't need to happen in this PR)

File: sources/jira-source/resources/spec.json (hunk 1 above)
Comment (ypc-faros) [reply sources/jira-source/resources/spec.json:257]: `use_faros_graph_for_board_selection` ?

File: sources/jira-source/src/project-board-filter.ts (hunk 2 above)
Comment (dbruno21) [reply sources/jira-source/src/project-board-filter.ts:190]: I'll try to generalize it when redoing it for GitHub

File: sources/jira-source/resources/spec.json (hunk 1 above)
Comment (dbruno21) [reply sources/jira-source/resources/spec.json:257]: @chalenge Any thoughts? Would you prefer a toogle / boolean instead?

File: sources/jira-source/resources/spec.json [hunk 3]
```diff
@@ -108,8 +108,15 @@
         "title": "Use Users Prefix Search",
         "default": false
       },
-      "projects": {
+      "use_faros_graph_boards_selection": {
         "order": 15,
+        "type": "boolean",
+        "title": "Use Faros Graph for Boards selection",
+        "description": "Query mode for selecting which Jira boards should be fetched (Faros credentials are required).",
```
Comment (ypc-faros) [inline sources/jira-source/resources/spec.json:115]: Update description

**Review by ypc-faros — APPROVED** [review]

**Pull request merged**